tests/tests_cluster_basic_custom_pcsd_tls.yml

```
@@ -69,3 +76,9 @@
                 == stat_pcsd_key_expected.stat.checksum
 
       tags: tests::verify
+
+      always:
+        - name: Clean up work directory
+          file:
+            path: "{{ __ha_cluster_work_dir.path }}"
+            state: absent
```

```suggestion
            state: absent
          delegate_to: localhost
          run_once: true
```
Clean the directory on the controller where it actually is created.